website/docs/r/data_lake_pipeline.html.markdown

```diff
@@ -0,0 +1,134 @@
+---
+layout: "mongodbatlas"
+page_title: "MongoDB Atlas: data_lake"
+sidebar_current: "docs-mongodbatlas-resource-data-lake"
+description: |-
+    Provides a Data Lake Pipeline resource.
+---
+
+# Resource: mongodbatlas_data_lake_pipeline
+
+`mongodbatlas_data_lake_pipeline` provides a Data Lake Pipeline resource.
+
+-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.
+
+~> **IMPORTANT:** All arguments including the password will be stored in the raw state as plain-text. [Read more about sensitive data in state.](https://www.terraform.io/docs/state/sensitive-data.html)
+
+## Example Usages
+
+
+```terraform
+resource "mongodbatlas_project" "projectTest" {
+  name   = "NAME OF THE PROJECT"
+  org_id = "ORGANIZATION ID"
+}
+
+resource "mongodbatlas_cluster" "automated_backup_test" {
+    project_id   = "63f4d4a47baeac59406dc131"
+    name         = "automated-backup-test"
+
+    provider_name               = "GCP"
+    provider_region_name        = "US_EAST_4"
+    provider_instance_size_name = "M10"
+    cloud_backup                = true   // enable cloud backup snapshots
+    mongo_db_major_version      = "6.0"
+  }
+
+
+resource "mongodbatlas_data_lake_pipeline" "pipeline" {
+      project_id       = mongodbatlas_project.projectTest.project_id
+      name       = "DataLakePipelineName"
+      sink {
+        type = "DLS"
+        partition_fields {
+            name = "access"
+            order = 0
+        }
+      }
+
+      source {
+        type = "ON_DEMAND_CPS"
+        cluster_name = mongodbatlas_cluster.automated_backup_test.name
+        database_name = "sample_airbnb"
+        collection_name = "listingsAndReviews"
+      }
+
+      transformations {
+              field = "test"
+              type  = "EXCLUDE"
+      }
+
+      transformations {
+              field = "test22"
+              type  = "EXCLUDE"
+      }
+}
+```
+
+## Argument Reference
+
+* `project_id` - (Required) The unique ID for the project to create a data lake pipeline.
+* `name` - (Required) Name of the Atlas Data Lake Pipeline.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` -  Unique 24-hexadecimal digit string that identifies the Data Lake Pipeline.
+* `created_date` - Timestamp that indicates when the Data Lake Pipeline was created.
+* `last_updated_date` - Timestamp that indicates the last time that the Data Lake Pipeline was updated.
+* `state` - State of this Data Lake Pipeline.
+* `sink` - Ingestion destination of a Data Lake Pipeline.
+  * `sink.0.type` - Type of ingestion destination of this Data Lake Pipeline.
```

Comment:
agree and helpful. effectively since sink only has max of type of item, in our docs we can replace sink.0.provider, sink.0.region, etc. with sink_provider, sink_region. current docs format is helpful when we have a TypeLists with multiple types of inputs in which in those cases no change is needed. if you are aligned @andreaangiolillo and @maastha can we adopt this format for all PRs going forward as well as get corrected in PR that has already been merged (see [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1163/files#diff-ebd5e08c7fa8e4f0fee0f3b1c8c687c52fd98ce77434a1053e591ee3952d7902R91))?

Reply:
@andreaangiolillo can we update the development guide/CONTRIBUTING.md with an example for array types best practice - we will hit this again.

Reply:
Sure. I have already updated the [development guide](https://docs.google.com/document/d/1QNNtzrDlhFC6avkjLWnBDtLFOikte2WV-UlcbXcQvS8/edit#bookmark=id.xssml9nw3md4). I created this [ticket](https://jira.mongodb.org/browse/INTMDB-840) to update the CONTRIBUTING.md

Reply:
PR is ready for your review https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1186